go-update/src/main/java/org/reactome/release/goupdate/DuplicateCleaner.java

```diff
@@ -0,0 +1,182 @@
+						// check each atttribute
+						for (GKSchemaAttribute refAtt : refAttribs)
+						{
+							@SuppressWarnings("unchecked")
+							Collection<GKInstance> refs = (Collection<GKInstance>) inst.getReferers(refAtt);
+							if (refs != null && refs.size() > 0)
+							{
+								for (GKInstance ref : refs)
+								{
+									if (!Arrays.asList(ReactomeJavaConstants.GO_BiologicalProcess, ReactomeJavaConstants.GO_MolecularFunction, ReactomeJavaConstants.GO_CellularComponent, ReactomeJavaConstants.Compartment).contains(ref.getSchemClass().getName()))
+									{
+										logger.info("\t\tvia {}:\t{}", refAtt.getName(), ref.toString());
+									}
+								}
+							}
+						}
+					}
+					// if there are NO referrers, add the db_id to the list of DB IDs without referrers.
+					else
+					{
+						logger.debug("DB ID {} for accession {} has no significant referrers, and will be deleted.", dbId, accession);
+						dbIDsWithNoReferrers.add(dbId);
+					}
+				}
+			}
+			// update list of things to delete with IDs that have no referrers.
+			dbIDsToDelete.addAll(dbIDsWithNoReferrers);
+			// If no DB IDs have significant referrers, we should delete ALL DB_IDs for the accession, except for the newest one.
+			if (dbIDsWithNoReferrers.size() == refCounts.keySet().size())
+			{
+				GKInstance newestInstance = null;
+				LocalDateTime newestDate = null;
+				DateTimeFormatter formatter = DateTimeFormatter.ofPattern("yyyy-MM-dd HH:mm:ss.S");
+				// Need to try all classes since we don't know which class a GO Accession might be.
+				for (String reactomeClass : Arrays.asList(ReactomeJavaConstants.GO_BiologicalProcess, ReactomeJavaConstants.GO_MolecularFunction, ReactomeJavaConstants.GO_CellularComponent))
+				{
+					GKInstance modifiedInstance = null;
+					
+					@SuppressWarnings("unchecked")
+					Collection<GKInstance> instances = (Collection<GKInstance>) adaptor.fetchInstanceByAttribute(reactomeClass, ReactomeJavaConstants.accession, "=", accession);
+					// For each instance for this accession, try to find the newest one so that it won't be deleted.
+					if (instances != null && instances.size() > 0)
+					{
+						for (GKInstance instance : instances)
+						{
+							dbIDsToDelete.add(instance.getDBID());
+							LocalDateTime localDate = null;
+							modifiedInstance = (GKInstance) instance.getAttributeValue(ReactomeJavaConstants.modified);
+							
+							if (modifiedInstance != null)
+							{
+								localDate = LocalDateTime.parse((CharSequence) modifiedInstance.getAttributeValue(ReactomeJavaConstants.dateTime), formatter);
+							}
+							else
+							{
+								modifiedInstance = (GKInstance) instance.getAttributeValue(ReactomeJavaConstants.created);
+								localDate = LocalDateTime.parse((CharSequence) modifiedInstance.getAttributeValue(ReactomeJavaConstants.dateTime), formatter);
+							}
+
+							if (newestInstance == null)
```

Comment:
Since they set the same variables and the same values, this check can be combined with next:
```
if (newestInstance == null || localDate.compareTo(newestDate) > 0) {
	newestInstance = instance;
	newestDate = localDate;
}
```